src/rules.jl

```diff
@@ -85,7 +85,7 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
 @define_diffrule Base.atan(x, y)    = :( $y / ($x^2 + $y^2)                                 ), :( -$x / ($x^2 + $y^2)                                                     )
 @define_diffrule Base.hypot(x, y)  = :( $x / hypot($x, $y)                                      ), :(  $y / hypot($x, $y)                                                     )
 @define_diffrule Base.log(b, x)    = :( log($x) * inv(-log($b)^2 * $b)                          ), :( inv($x) / log($b)                                                       )
-@define_diffrule Base.ldexp(x, y)  = :( exp2($y)                                                ), :NaN
+@define_diffrule Base.ldexp(x, y)  = :( oftype($x, exp2($y))                                                ), :(oftype(float($x), NaN))
```

Comment:
I don't think `NaN` has to be converted here?
```suggestion
@define_diffrule Base.ldexp(x, y)  = :( oftype($x, exp2($y))                                                ), :NaN
```

Reply:
Sorry I missed that one!